packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2596,6 +2606,15 @@ export class TransactionController extends BaseController<
     });
   }
 
+  #addBatchMetadata(transactionBatchMeta: TransactionBatchMeta) {
```

Comment:
In the interest of modularity, could we provide the `update` callback directly so the state is encapsulated within the `batch` module?

Reply:
Removed `#addBatchMetadata` as we agreed to not persist the `transactionBatches` for now.

Reply:
Sorry for the confusion, we need to update the state while the confirmation is active, so the client can obtain the information via the ID.

But we can delete again once finished, I meant "persist" as in save in the state long term.